### ISSUE TITLE
EntryPointNotFoundException after upgrading to Azure SDK 2.6

### ISSUE THREAD

**Issue by maiorfi** [dotnet/orleans#424]
It throws after having updated CloudService project to 2.6 SDK, in:

_orleansAzureSilo.Start(RoleEnvironment.DeploymentId, RoleEnvironment.CurrentRoleInstance);

This is the full call stack:

Microsoft.WindowsAzure.ServiceRuntime Critical: 1 : Unhandled Exception:
System.EntryPointNotFoundException: Punto di ingresso non trovato.
   in System.Collections.Generic.IEnumerable`1.GetEnumerator()
   in System.String.Join[T](String separator, IEnumerable`1 values)
   in Orleans.Runtime.Host.AzureSilo.GetEndpointInfo(RoleInstance roleInstance, String endpointName)
   in Orleans.Runtime.Host.AzureSilo.Start(String deploymentId, RoleInstance myRoleInstance, ClusterConfiguration config)
   in Orleans.Runtime.Host.AzureSilo.Start(String deploymentId, RoleInstance myRoleInstance)
   in CloudOrleansSilo.WorkerRole.Run() in c:\Progetti\DemoVari\orleans\SistemaGestioneNavette\CloudOrleansSilo\WorkerRole.cs:riga 115
   in Microsoft.WindowsAzure.ServiceRuntime.RoleEnvironment.StartRoleInternal()
   in Microsoft.WindowsAzure.ServiceRuntime.RoleEnvironment.StartRole()
   in Microsoft.WindowsAzure.ServiceRuntime.Implementation.Loader.RoleRuntimeBridge.<StartRole>b__2()
   in System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   in System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   in System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   in System.Threading.ThreadHelper.ThreadStart()

Any hint?

**Comment by gabikliot** [comment]
Most probably you are missing binding redirects for 2.6: http://dotnet.github.io/orleans/Deployment-and-Versioning-of-Dependencies

**Comment by ca-ta** [comment]
@maiorfi: Here's what works for me, hope it helps:

```
[...]
      <dependentAssembly>
        <assemblyIdentity name="WindowsAzureTelemetryEvents" culture="neutral" publicKeyToken="31bf3856ad364e35" />
        <bindingRedirect oldVersion="0.0.0.0-2.6.0.0" newVersion="2.6.0.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="WindowsAzureEventSource" culture="neutral" publicKeyToken="31bf3856ad364e35" />
        <bindingRedirect oldVersion="0.0.0.0-2.6.0.0" newVersion="2.6.0.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="msshrtmi" culture="neutral" publicKeyToken="31bf3856ad364e35" />
        <bindingRedirect oldVersion="0.0.0.0-2.6.0.0" newVersion="2.6.0.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.WindowsAzure.ServiceRuntime" culture="neutral" publicKeyToken="31bf3856ad364e35" />
        <bindingRedirect oldVersion="0.0.0.0-2.6.0.0" newVersion="2.6.0.0" />
      </dependentAssembly>
[...]
```

**Comment by pherbel** [comment]
I think the main problem is the Microsoft.WindowsAzure.ServiceRuntime.dll which is coming from the Unofficial.Microsoft.WindowsAzure.ServiceRuntime 2.5.0.0 nugget package.

When you add the Microsoft.Orleans.AzureUtils nuget package it will override the current 2.6 version (SDK) with the 2.5 version (unofficial nugget package).

Unfortunately currently there is no "Unofficial" 2.6 nuget package so you need to use the dll from the SDK directory. (It would be great if Microsoft creates an official package from this)

Solution:
Add back the Microsoft.WindowsAzure.ServiceRuntime.dll 2.6 from the SDK directory and create a binding redirects.
C:\Program Files\Microsoft SDKs\Azure\.NET SDK\v2.6\ref\Microsoft.WindowsAzure.ServiceRuntime.dll

I hope this helps

**Comment by gabikliot** [comment]
Looks like this was fixed with 1.0.9 release, where we eliminated dependency on specific ServiceRuntime version.
